src/daft-dsl/src/expr/mod.rs

```diff
@@ -1125,7 +1125,7 @@ impl Expr {
                 _ => inputs.first().unwrap().name(),
             },
             Self::ScalarFunction(func) => match func.name() {
-                "to_struct" => "struct", // FIXME: make .name() use output name from schema
+                "struct" => "struct", // FIXME: make struct its own expr variant
```

Comment:
Did we also need to keep `"to_struct" => "struct"` for backward compat here? Also ok with this change given that SQL is less stable of an API right now compared to dataframe